livekit-rtc/livekit/rtc/participant.py

```diff
@@ -539,6 +549,119 @@ async def set_attributes(self, attributes: dict[str, str]) -> None:
         finally:
             FfiClient.instance.queue.unsubscribe(queue)
 
+    async def stream_text(
+        self,
+        *,
+        destination_identities: List[str] = [],
+        topic: str = "",
+        extensions: Dict[str, str] = {},
+        reply_to_id: str | None = None,
+        total_size: int | None = None,
+    ) -> TextStreamWriter:
+        """
+        Returns a TextStreamWriter that allows to write individual chunks of text to a text stream.
+        In most cases where you want to simply send a text message use send_text() instead.
+        """
+        writer = TextStreamWriter(
+            self,
+            topic=topic,
+            extensions=extensions,
+            reply_to_id=reply_to_id,
+            destination_identities=destination_identities,
+            total_size=total_size,
+        )
+
+        await writer._send_header()
+
+        return writer
+
+    async def send_text(
+        self,
+        text: str,
+        *,
+        destination_identities: List[str] = [],
+        topic: str = "",
+        extensions: Dict[str, str] = {},
+        reply_to_id: str | None = None,
+    ):
+        total_size = len(text.encode())
+        writer = await self.stream_text(
+            destination_identities=destination_identities,
+            topic=topic,
+            extensions=extensions,
+            reply_to_id=reply_to_id,
+            total_size=total_size,
+        )
+
+        for chunk in split_utf8(text, STREAM_CHUNK_SIZE):
+            await writer.write(chunk)
+        await writer.aclose()
+
+        return writer.info
+
+    async def stream_bytes(
+        self,
+        name: str,
+        *,
+        total_size: int | None = None,
+        mime_type: str = "application/octet-stream",
+        extensions: Optional[Dict[str, str]] = None,
+        stream_id: str | None = None,
+        destination_identities: Optional[List[str]] = None,
+        topic: str = "",
+    ) -> ByteStreamWriter:
+        """
+        Returns a ByteStreamWriter that allows to write individual chunks of bytes to a file stream.
+        In cases where you want to simply send a file from the file system use send_file() instead.
+        """
+        writer = ByteStreamWriter(
+            self,
+            name=name,
+            extensions=extensions,
+            total_size=total_size,
+            stream_id=stream_id,
+            mime_type=mime_type,
+            destination_identities=destination_identities,
+            topic=topic,
+        )
+
+        await writer._send_header()
+
+        return writer
+
+    async def send_file(
+        self,
+        file_path: str,
+        topic: str = "",
+        destination_identities: Optional[List[str]] = None,
```

Comment:
would be great to add `file_name` as an optional kwarg.

picking a default filename is great.. but it's also helpful if I want a local tempfile to be sent, and don't know/care about the actual filename on disk

Reply:
optional `name` is fine with me, but considering the fact that we now went from naming it `file_name` to `name` on the receiving side, I don't think we should call it `file_name` here either, wdyt?

Reply:
sure!